pkg/app/mysql-deploymentconfig.go

```diff
@@ -0,0 +1,275 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/kube"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/openshift"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	osversioned "github.com/openshift/client-go/apps/clientset/versioned"
+)
+
+const (
+	mysqlDepConfigWaitTimeout = 1 * time.Minute
+	mysqlToolPodName          = "mysql-tools-pod"
+	mysqlToolContainerName    = "mysql-tools-container"
+	mysqlToolImage            = "kanisterio/mysql-sidecar:0.24.0"
+)
+
+type MysqlDepConfig struct {
+	cli        kubernetes.Interface
+	osCli      osversioned.Interface
+	name       string
+	namespace  string
+	osAppImage string
+	envVar     map[string]string
+}
+
+func NewMysqlDepConfig(name string) App {
+	return &MysqlDepConfig{
+		name:       name,
+		osAppImage: "openshift/mysql-56-centos7",
+		envVar: map[string]string{
+			"MYSQL_ROOT_PASSWORD": "secretpassword",
+		},
+	}
+}
+
+func (mdep *MysqlDepConfig) Init(context.Context) error {
+	cfg, err := kube.LoadConfig()
+	if err != nil {
+		return err
+	}
+
+	mdep.cli, err = kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	mdep.osCli, err = osversioned.NewForConfig(cfg)
+
+	return err
+}
+
+func (mdep *MysqlDepConfig) Install(ctx context.Context, namespace string) error {
+	mdep.namespace = namespace
+
+	oc := openshift.NewOpenShiftClient()
+	out, err := oc.NewApp(ctx, mdep.namespace, mdep.osAppImage, mdep.envVar)
+	if err != nil {
+		return errors.Wrapf(err, "Error installing app on openshift cluster.", field.M{"app": mdep.name, "out": out})
+	}
+
+	err = mdep.createMySQLToolsPod(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "Error creating mysql tools pod")
+	}
+
+	err = mdep.createMySQLSecret(ctx)
+	return err
+}
+
+// createMySQLSecret creates a secret that will be used to login to running MYSQL instance
+func (mdep *MysqlDepConfig) createMySQLSecret(ctx context.Context) error {
+	mysqlSecret := &v1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mdep.name,
+			Namespace: mdep.namespace,
+		},
+		Data: map[string][]byte{
+			"mysql-root-password": []byte(mdep.envVar["MYSQL_ROOT_PASSWORD"]),
+		},
+	}
+
+	_, err := mdep.cli.CoreV1().Secrets(mdep.namespace).Create(mysqlSecret)
+	if err != nil {
```

Comment:
No need to check if err is nil. Same at other places

Reply:
I wanted to wrap the error detail with the actual message, do you think removing that is better option?